core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala

```diff
@@ -150,4 +163,228 @@ class DescribeConsumerGroupTest extends KafkaServerTestHarness {
     consumer1Mock.stop()
     consumer2Mock.stop()
   }
+
+  @Test
+  def testDescribeNonExistingGroupWithNewConsumer() {
+    // run one consumer in the group consuming from a single-partition topic
+    val executor = new ConsumerGroupExecutor(brokerList, 1, group, topic)
+
+    // note the group to be queried is a different (non-existing) group
+    val cgcArgs = Array("--bootstrap-server", brokerList, "--describe", "--group", "missing.group")
+    val opts = new ConsumerGroupCommandOptions(cgcArgs)
+    val consumerGroupCommand = new KafkaConsumerGroupService(opts)
+
+    TestUtils.waitUntilTrue(() => {
+        try {
+          val (state, assignments) = consumerGroupCommand.describeGroup()
+          state == Some("Dead") &&
+          assignments == Some(List())
+        } catch {
+          case _: GroupCoordinatorNotAvailableException | _: IllegalArgumentException =>
+            // Do nothing while the group initializes
+            false
+          case e: Throwable =>
+            throw e
+            true
```

Comment:
nit: pretty sure we shouldn't need this if we're throwing an exception above. More of these below.